src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs

```diff
@@ -521,14 +515,53 @@ async Task<InstanceStatus> FetchInstanceStatusInternalAsync(string instanceId, s
                 executionId ?? string.Empty,
                 stopwatch.ElapsedMilliseconds);
 
-            OrchestrationInstanceStatus orchestrationInstanceStatus = (OrchestrationInstanceStatus)orchestration.Result;
-            if (orchestrationInstanceStatus == null)
+            var tableEntity = segment.ToList().FirstOrDefault();
+            if (tableEntity == null)
             {
                 return null;
             }
 
-            OrchestrationState state = await this.ConvertFromAsync(orchestrationInstanceStatus, instanceId);
-            return state != null ? new InstanceStatus(state, orchestration.Etag) : null;
+            this.stats.TableEntitiesRead.Increment(1);
+
+            var orchestrationState = await this.ConvertFromAsync(tableEntity);
+
+            return new InstanceStatus(orchestrationState, tableEntity.ETag);
+        }
+
+        Task<OrchestrationState> ConvertFromAsync(DynamicTableEntity tableEntity)
+        {
+            var properties = tableEntity.Properties;
+            var orchestrationInstanceStatus = ConvertFromAsync(properties);
+
+            return ConvertFromAsync(orchestrationInstanceStatus, tableEntity.PartitionKey);
+        }
+
+        private OrchestrationInstanceStatus ConvertFromAsync(IDictionary<string, EntityProperty> properties)
```

Comment:
nit: make this `static` since it doesn't reference any instance state. That helps the reader understand that the method doesn't depend on external instance state and also gives it a small performance boost.

```suggestion
        static OrchestrationInstanceStatus ConvertFromAsync(IDictionary<string, EntityProperty> properties)
```